src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Subscriber.kt

```diff
@@ -116,18 +115,28 @@ class TaskReceiver(tasksDirectory: String,
         if (!rootFile.exists()) {
             throw IOException("In task(id = $taskId): path to root file doesn't exists")
         }
-      
+
         return rootFile
     }
 
+    private fun compileProject(request: CompilerFleet.CompilerFleetRequest): File {
+        val rootFileFullPath = request.rootFileName
+        val zippedProject = request.zipBytes
+        val engine = request.engine
+
+        if (request.compiler == CompilerFleet.Compiler.MOCK) {
+            return MOCK_PDF_FILE
+        }
+
+        // TODO: Unzip project and compile via docker
```

Comment:
Maybe just return that previous code? dockerProcessor, etc?

Reply:
OK, can I unzip project in request using my function?
This is just collection of files, right?